x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/DeclarativeQueryTranslatorTests.java

```diff
@@ -0,0 +1,216 @@
+                    sb.append(line).append(' ');
+                    if (line.endsWith(";")) {
+                        sb.setLength(sb.length() - 2);
+                        query = sb.toString();
+                        sb.setLength(0);
+                    }
+                }
+
+                else {
+                    if (line.endsWith(";")) {
+                        line = line.substring(0, line.length() - 1);
+                        done = true;
+                    }
+
+                    // no expectation
+                    if (line.equals("null") == false && line.isEmpty() == false) {
+                        String[] matcherAndExpectation = line.split("[ \\t]+", 2);
+                        if (matcherAndExpectation.length == 1) {
+                            matchers.add(containsString(matcherAndExpectation[0]));
+                        } else if (matcherAndExpectation.length == 2) {
+                            String matcherType = matcherAndExpectation[0];
+                            String expectation = matcherAndExpectation[1];
+                            switch (matcherType.toUpperCase(Locale.ROOT)) {
+                                case MATCHER_TYPE_CONTAINS:
+                                    matchers.add(containsString(expectation));
+                                    break;
+                                case MATCHER_TYPE_REGEX:
+                                    matchers.add(containsRegex(expectation));
+                                    break;
+                                default:
+                                    throw new IllegalArgumentException("unsupported matcher on line "
+                                        + TEST_FILENAME + ":" + lineNumber + ": " + matcherType);
+                            }
+                        } else {
+                            throw new IllegalArgumentException("wrongly formatted assert, " +
+                                "should be '<regex>? <expectation>', but on line "
+                                + TEST_FILENAME + ":" + lineNumber + " it was: " + line);
+                        }
+                    }
+
+                    if (done) {
+                        // Add and zero out for the next spec
+                        addSpec(arr, name, query, matchers);
+                        name = null;
+                        query = null;
+                        matchers = new ArrayList<>(8);
+                        done = false;
+                    }
+                }
+            }
+
+            if (name != null) {
+                throw new IllegalStateException("Read a test [" + name + "] without a body at the end of [" + TEST_FILENAME + "]");
+            }
+        }
+        return arr;
+    }
+
+    private static void addSpec(ArrayList<Object[]> arr, String name, String query, 
+        List<Matcher<String>> matchers) {
+        if ((Strings.isNullOrEmpty(name) == false) && (Strings.isNullOrEmpty(query) == false)) {
```

Comment:
Seems like `name` will never match the condition and `query` can't be `null`. It might actually be a good idea to check that's also not empty before setting it and the whole `if` could be done away with (maybe also the function?).